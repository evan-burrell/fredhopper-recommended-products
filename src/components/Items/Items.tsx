import React from 'react';
import Slider from 'react-slick';
import { SLIDER_SETTINGS } from '../../constants';

const Items = ({ list }: ListProps): React.ReactElement => (
  <Slider {...SLIDER_SETTINGS}>
    {list.map(({ secondid, _imageurl, title }) => (
      <div key={secondid}>
        <div className="c-product-carousel__item">
          <div className="c-product-carousel__item-image-container">
            <img
              className="c-product-carousel__item-image"
              src={_imageurl}
              alt=""
            />
          </div>
          <h3 className="c-product-carousel__item-title">{title}</h3>
          <p className="c-product-carousel__item-price">Â£48.00</p>
        </div>
      </div>
    ))}
  </Slider>
);

export default Items;
