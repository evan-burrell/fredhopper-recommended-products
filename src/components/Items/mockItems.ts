export const mockItems = [
  {
    secondid: '1',
    brand: 'brand1',
    _imageurl: 'image.url',
    vg_url_key: 'localhost',
    url_key: '',
    title: 'I am a title am a title am a title am a title'
  },
  {
    secondid: '2',
    brand: 'brand2',
    _imageurl: 'image.url',
    vg_url_key: 'localhost',
    url_key: '',
    title:
      'I am a title am a title am a title am a title I am a title am a title am a title am a title I am a title am a title am a title am a title I am a title am a title am a title am a title'
  },
  {
    secondid: '3',
    brand: 'brand3',
    _imageurl: 'image.url',
    vg_url_key: 'localhost',
    url_key: '',
    title: 'I am a title am a title am a title am a title'
  },
  {
    secondid: '4',
    brand: 'brand4',
    _imageurl: 'image.url',
    vg_url_key: 'localhost',
    url_key: '',
    title: 'I am a title am a title am a title am a title'
  },
  {
    secondid: '5',
    brand: 'brand5',
    _imageurl: 'image.url',
    vg_url_key: 'localhost',
    url_key: '',
    title: 'I am a title am a title am a title am a title'
  },
  {
    secondid: '6',
    brand: 'brand6',
    _imageurl: 'image.url',
    vg_url_key: 'localhost',
    url_key: '',
    title:
      'I am a title am a title am a title am a title I am a title am a title am a title am a title I am a title am a title am a title am a title I am a title am a title am a title am a title'
  },
  {
    secondid: '7',
    brand: 'brand7',
    _imageurl: 'image.url',
    vg_url_key: 'localhost',
    url_key: '',
    title: 'I am a title am a title am a title am a title'
  },
  {
    secondid: '8',
    brand: 'brand8',
    _imageurl: 'image.url',
    vg_url_key: 'localhost',
    url_key: '',
    title: 'I am a title am a title am a title am a title'
  }
];
