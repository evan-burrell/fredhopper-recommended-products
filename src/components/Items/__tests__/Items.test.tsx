import React from 'react';
import { render } from '@testing-library/react';
import Items from '../Items';
import { mockItems } from '../mockItems';

describe('Items', () => {
  const { container } = render(<Items list={mockItems} />);

  it('matches the rendered snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
