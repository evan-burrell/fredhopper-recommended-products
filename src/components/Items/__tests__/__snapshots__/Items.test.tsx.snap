// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items matches the rendered snapshot 1`] = `
<div>
  <div
    class="slick-slider slick-initialized"
    dir="ltr"
  >
    <button
      class="slick-arrow slick-prev slick-disabled"
      data-role="none"
      style="display: block;"
      type="button"
    >
       
      Previous
    </button>
    <div
      class="slick-list"
    >
      <div
        class="slick-track"
        style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
      >
        <div
          aria-hidden="false"
          class="slick-slide slick-active slick-current"
          data-index="0"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <div
                class="c-product-carousel__item"
              >
                <div
                  class="c-product-carousel__item-image-container"
                >
                  <img
                    alt=""
                    class="c-product-carousel__item-image"
                    src="https://secretsales-prod-product-images.s3-eu-west-1.amazonaws.com/686175/2af147f69c072085d3ac53b14296e1fa.jpg"
                  />
                </div>
                <h3
                  class="c-product-carousel__item-title"
                >
                  I am a title am a title am a title am a title
                </h3>
                <p
                  class="c-product-carousel__item-price"
                >
                  £48.00
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="slick-slide slick-active"
          data-index="1"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <div
                class="c-product-carousel__item"
              >
                <div
                  class="c-product-carousel__item-image-container"
                >
                  <img
                    alt=""
                    class="c-product-carousel__item-image"
                    src="https://www.brodi.com/image/cache/data/color_swatch/5-100x100.jpg"
                  />
                </div>
                <h3
                  class="c-product-carousel__item-title"
                >
                  I am a title am a title am a title am a title I am a title am a title am a title am a title I am a title am a title am a title am a title I am a title am a title am a title am a title
                </h3>
                <p
                  class="c-product-carousel__item-price"
                >
                  £48.00
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="slick-slide slick-active"
          data-index="2"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <div
                class="c-product-carousel__item"
              >
                <div
                  class="c-product-carousel__item-image-container"
                >
                  <img
                    alt=""
                    class="c-product-carousel__item-image"
                    src="https://secretsales-prod-product-images.s3-eu-west-1.amazonaws.com/649894/8d2af934b2df22f5be7d4cc3c08497a3.jpg"
                  />
                </div>
                <h3
                  class="c-product-carousel__item-title"
                >
                  I am a title am a title am a title am a title
                </h3>
                <p
                  class="c-product-carousel__item-price"
                >
                  £48.00
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="slick-slide slick-active"
          data-index="3"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <div
                class="c-product-carousel__item"
              >
                <div
                  class="c-product-carousel__item-image-container"
                >
                  <img
                    alt=""
                    class="c-product-carousel__item-image"
                    src="https://secretsales-prod-product-images.s3-eu-west-1.amazonaws.com/662569/586517eb71a571d3f3a56bbdb7b7d015.jpg"
                  />
                </div>
                <h3
                  class="c-product-carousel__item-title"
                >
                  I am a title am a title am a title am a title
                </h3>
                <p
                  class="c-product-carousel__item-price"
                >
                  £48.00
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide"
          data-index="4"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <div
                class="c-product-carousel__item"
              >
                <div
                  class="c-product-carousel__item-image-container"
                >
                  <img
                    alt=""
                    class="c-product-carousel__item-image"
                    src="https://secretsales-prod-product-images.s3-eu-west-1.amazonaws.com/686175/2af147f69c072085d3ac53b14296e1fa.jpg"
                  />
                </div>
                <h3
                  class="c-product-carousel__item-title"
                >
                  I am a title am a title am a title am a title
                </h3>
                <p
                  class="c-product-carousel__item-price"
                >
                  £48.00
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide"
          data-index="5"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <div
                class="c-product-carousel__item"
              >
                <div
                  class="c-product-carousel__item-image-container"
                >
                  <img
                    alt=""
                    class="c-product-carousel__item-image"
                    src="https://www.brodi.com/image/cache/data/color_swatch/5-100x100.jpg"
                  />
                </div>
                <h3
                  class="c-product-carousel__item-title"
                >
                  I am a title am a title am a title am a title I am a title am a title am a title am a title I am a title am a title am a title am a title I am a title am a title am a title am a title
                </h3>
                <p
                  class="c-product-carousel__item-price"
                >
                  £48.00
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide"
          data-index="6"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <div
                class="c-product-carousel__item"
              >
                <div
                  class="c-product-carousel__item-image-container"
                >
                  <img
                    alt=""
                    class="c-product-carousel__item-image"
                    src="https://secretsales-prod-product-images.s3-eu-west-1.amazonaws.com/649894/8d2af934b2df22f5be7d4cc3c08497a3.jpg"
                  />
                </div>
                <h3
                  class="c-product-carousel__item-title"
                >
                  I am a title am a title am a title am a title
                </h3>
                <p
                  class="c-product-carousel__item-price"
                >
                  £48.00
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide"
          data-index="7"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <div
                class="c-product-carousel__item"
              >
                <div
                  class="c-product-carousel__item-image-container"
                >
                  <img
                    alt=""
                    class="c-product-carousel__item-image"
                    src="https://secretsales-prod-product-images.s3-eu-west-1.amazonaws.com/662569/586517eb71a571d3f3a56bbdb7b7d015.jpg"
                  />
                </div>
                <h3
                  class="c-product-carousel__item-title"
                >
                  I am a title am a title am a title am a title
                </h3>
                <p
                  class="c-product-carousel__item-price"
                >
                  £48.00
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="slick-arrow slick-next"
      data-role="none"
      style="display: block;"
      type="button"
    >
       
      Next
    </button>
  </div>
</div>
`;
